# Exam 2017

In [2]:
library(tidyverse) # readr, ggplot, dplyr, ...
library(gridExtra) # Arrangning ggplots
library(readxl)    # Loading excel
library(repr)      # Resizing the plots in jupyter
library(rugarch)   # For ARCH and GARCH models
library(dynlm)     # Lagged linear models
library(urca)      # Dick Fuller test
library(car)       # Linear hyphothesis testing
library(tseries)   # adf-test
library(vars)      # VAR models
library(zoo)       # Convert quarters to date
library(forecast)  

options(repr.plot.width=5, repr.plot.height=2.5)

In [6]:
setwd("C:/Users/larsn/Documents/GitHub/Applied-Econometric-Time-Series-WE")

data1 <- read_csv("Data/exam/data1.csv")
head(data1)

data2 <- read_csv("Data/exam/data2.csv")
head(data2)

kointegration <- read_csv("Data/exam/kointegration.csv")
head(kointegration)

Parsed with column specification:
cols(
  x = col_character()
)


x            
1 2.344.394.908
2 295.853.768  
3 2.006.881.046
4 0.936106173  
5 1.714.408.477
6 2.741.346.693

Parsed with column specification:
cols(
  `date;GDPdeflator;pi` = col_character()
)


date;GDPdeflator;pi         
1 Q4 1959;17.375;             
2 Q1 1960;17.409;0.78196897   
3 Q2 1960;17.473;1.467.807.392
4 Q3 1960;17.551;1.781.638.395
5 Q4 1960;17.63;1.796.427.226 
6 Q1 1961;17.651;0.476177035

Parsed with column specification:
cols(
  `date (YYYYQQ);c;a;y` = col_character()
)


date (YYYYQQ);c;a;y                             
1 195104;8.945.843.412;1.056.260.837;8.658.345.436
2 195201;8.944.092.233;1.057.123.571;866.592.584  
3 195202;8.957.498.884;1.058.301.045;8.667.349.937
4 195203;8.966.563.493;1.057.800.623;8.681.577.904
5 195204;897.540.805;1.057.487.284;8.710.945.649  
6 195301;8.985.109.507;1.057.435.704;8.721.970.747

**Spørgsmål 1.1** 
Betragt nedenstående proces. Hvilken type proces er der tale om? Husk at forklare alle symboler.

$$y_t = \mu + \varepsilon_t + \beta_1\varepsilon_{t-1} + \beta_2\varepsilon_{t-2} $$

MA(2) process

**Spørgsmål 1.2**
Skriv udtrykket ved hjælp af lag operator notationen.

$$y_t = \mu + \varepsilon_t  + \beta_1L\varepsilon + \beta_2L^2\varepsilon $$ 

**Spørgsmål 1.3** Udled teoretisk den ubetingede (unconditional) middelværdi og varians for denne proces.

**Spørgsmål 1.4**
Udled teoretisk autokorrelationsfunctionen (ACF) og den partielle autokorrelationsfunktion (PACF).

**Spørgsmål 1.5**
På baggrund af data i fanebladet "opgave1" i fillen benævnt dataExam2016.xlsx bedes du estimere og afbilde ACF og PACF. Hvilken type proces vurderer du der er tale om?

**Spørgsmål 1.6**
Antag at $\mu = 1, \beta_1 = 0.6, \beta_2 = 0.2, \sigma^2 = 1$ Sammenhold den analytiske ACF og PACF på baggrund af disse parametre med den estimerede i spørgsmålet ovenfor - er der overensstemmelse mellem disse? Hvorfor/hvorfor ikke?

# Tidsvarierende volatilitet [20%]
Denne opgave består af 2 dele. Den første del er overvejende teoretisk mens anden del i spørgsmål 2.5 har en mere implementerende karakter. Begge dele indgår ligeligt i vurderingen.

**Spørgsmål 2.1** Et eksempel på en økonometrisk model, der tillader tidsvarierende varians (tidsvarierende volatilitet) er GARCH modellen. Skriv de nødvendige ligninger op for en GARCH(1,1) model og redegør for, hvordan man kan se den tidsvarierende varians. Husk at beskrive symbolerne i ligningerne og eventuelle antagelser.

**Spørgsmål 2.2** Hvilke restriktioner på parametrene skal være opfyldt for at GARCH modellen er stationær?

**Spørgsmål 2.3** Hvis fejlleddet er beskrevet ved en GARCH(1,1) proces, hvilken proces følger de kvadrerede residuler "2 t da?

De følger en ARMA(1,1) model

**Spørgsmål 2.4** Hvordan vil du i praksis forsøge at vurdere om der er ARCH/GARCH effekter i fejlleddet "t; eksempelvis ved en simpel graf?

**Spørgsmål 2.5** Empirisk applikation. Brug data for infationen fra Excel filen benævnt dataExam2016 fra fanebladet "GARCH". Find en passende ARCH eller GARCH model for infationen for perioden 1960:Q1 til 2013:Q2. Afprøv maksimalt 4 ARCH/GARCH modeller og rapporter modellernes koefcientestimater, standard errors og informationskriterie(r). Plot også den estimerede volatilitet.

**Spørgsmål 2.6** Slut af med at teste om din foretrukne ARCH/GARCH model er velspecificeret.

# Opgave 3: Kointegration og fejlkorrektionsmodeller [45%]
I en ind‡ydelsesrig artikel1 foreslog Lettau og Ludvigson at forbrug, formue og indkomst må udgøre en langsigtet ligevægt. Forbrugeren ønsker at udjævne forbruget, og med udsigt til forøgede fremtidige formueafkast sættes forbruget i vejret, mens det omvendte gør sig gældende ved udsigten til faldende afkast. Formue udgøres dog ikke alene af …- nansielle aktiver men også af human kapital, hvor udsigten til forøgede fremtidige afkast på human kapital får forbrugeren til at øge forbruget i dag. Forfatterne approksimerer human kapital som proportional til indkomsten. Hvis således forbruget (c) i dag er højt relativt til den …nansielle forme (a) og human kapitalen approksimeret ved indkomsten (y), så må der på lang sigt være en ligevægtsrelation mellem disse variable, ifølge Let- tau og Ludvigson. Din opgave er nu empirisk at veri…cere denne påstand ved hjælp af kointegrationsanalyse og fejlkorrektionsmodeller.

**Spørgsmål 3.1** Giv en formel formel de…nition på kointegration.

**Spørgsmål 3.2** Giv dernæst en intuitiv forklaring på hvorfor 2 eller ‡ere variable kan vise sig at være kointegreret. Max. 1
2 side.

**Spørgsmål 3.3** For hver af variablene c; a; og y bedes du ved et unit root test afgøre om det opgivne tidsrækkedata på variablene er I (1) eller I (0) : Data ses i fanebladet "Let- tauLudvigson"i dataExam2016.xlsx. Hvad er nulhypotesen i dit test? Skriv din endelige modelspeci…kation op og gør rede for dine overvejelser omkring din modelspeci…kation, herunder eventuelle øvrige statistiske test. Dit svar skal slutte af med en tabel, hvor læseren klart og tydeligt kan a‡æse konklusionen og de valg du har tru¤et.

**Spørgsmål 3.4** Ved hjælp afEngle-Granger metoden bedes du nu teste for kointegration mellem c; a; og y: Lad dit udgangspunkt være en langsigtet sammenhæng i form af:
ct = ?0 + ?aat + ?yyt + et (3.1)
hvor du igen slutter dit svar af med en tabel, hvor læseren klart og tydeligt kan a‡æse konklusionen og de valg du har tru¤et. Husk at rapportere kritiske værdier.

**Spørgsmål 3.5** Som en del af Engle-Granger metoden bedes du også estimere en fejl- korrektionsmodel for de tre variable. Kommentér på resultaterne.

**Spørgsmål 3.6** Johansen metoden er et alternativ til Engle-Granger. Hvad er fordelene ved Johansen metoden? Max. 1
2 side.

**Spørgsmål 3.7** Du får nu opgivet nedenstående resultater på baggrund af Johansen metoden. Du bedes argumentere for antallet af kointegrationsrelationer ifølge Johansen metoden. I dit svar bedes du skrive de anvendte test op og kommentere på nulhypotesen. Kommer du frem til en anden konklusion sammenlignet med Engle-Granger?

![Alt text](data/exam/table.png)

**Spørgsmål 3.8** Givet din konklusion på antallet af kointegrationsrelationer fra oven- stående spørgsmål, hvad er da rangen afden såkaldte ? matrice? Hvordan hænger rangen af ? sammen med egenværdierne?

**Spørgsmål 3.9** Konkludér kortfattet på, om du …nder empirisk belæg for Lettau og Ludvigsons postulerede sammenhæng mellem c; a og y:Maximalt 1
2 side

# 4 Vektorautoregressive modeller [25%]
Betragt en trivariate VAR(2) model med nedenstående estimerede parametre fra den reducerede form. Med trivariate menes 3 variable.


$$
\begin{bmatrix} y_{1t} \\ y_{2t} \\ y_{3t}  \end{bmatrix}
=
\begin{bmatrix} -0.50 & 0.20 & 1.80 \\
                0.10 & -0.25 & 0.60 \\
                -0.05 & 0.50 & -0.60 
\end{bmatrix}
\begin{bmatrix} y_{1,t-1} \\ y_{2,t-1} \\ y_{3,t-1}  \end{bmatrix}
+
\begin{bmatrix} -0.20 & 0.30 & 2.00 \\
                0.10 & 0.02 & -0.02 \\
                0.07 & 0.70 & -0.05 
\end{bmatrix}
\begin{bmatrix} y_{1,t-2} \\ y_{2,t-2} \\ y_{3,t-2}  \end{bmatrix}
+
\begin{bmatrix} \varepsilon_{1t} \\ \varepsilon_{2t} \\ \varepsilon_{3t}  \end{bmatrix}
$$

hvor vektoren $\varepsilon_t \sim N(0,\Sigma_\varepsilon)$ og med
$$
\Sigma_\varepsilon =
\begin{bmatrix} 0.0080 & 0.0002 & 0.0003 \\
                0.0002 & 0.0004 & 0.0001 \\
                0.0003 & 0.0001 & 0.0017 
\end{bmatrix}
$$

**Spørgsmål 4.1** Er modellen stationær? Redegør i detaljer for dit svar og anvend tallene ovenfor i dine beregninger. Tip: Hvis du ikke vil bruge R i dine beregninger kan du overve- je en online matrix calculator; eksempelvis http://www.bluebit.gr/matrix-calculator/calculate.aspx eller anden.

**Spørgsmål 4.2** De…nér begrebet ’impulse responses’ og hvad man kan bruge ’impulse response’ analyse til. Forklar hvorfor det ikke giver mening at anvende den reducerede form VAR i strukturel impulse response analyse.

**Spørgsmål 4.3** Hvor mange identi…cerende restriktioner må du pålægge den reducerede form VAR?

**Spørgsmål 4.4** Giv en diskussion på maksimalt 2 sider af hvorfor pålægger man iden- ti…cerende restriktioner. Du kan overveje at medtage et eksempel i din diskussion.

**Spørgsmål 4.5** Beregn en Cholesky dekomponering af?" og anvend denne i en impulse response beregning for de 3 variable over en horisont på mindst 3 perioder. Speci…kt bedes du:

1. Betragte VMA formen; speci…kt Ci rekursionen for de første i = 1; ::; 3 led i Yt = P1
i=0 Ci"t?i; hvor Yt angiver vektoren med de tre variable på tid t, dvs. Yt =
(y1t; y2t; y3t). 1 Du bedes eksplicit rapportere responsen for y2 efter 3 perioder som følge af et chok til den første variabel.

2. Betragte SVMA; specifikt $\Theta_i$ rekursionen for de første i = 1; ::; 3 led i Yt = P1
i=1 ?i"t?i; hvor Yt angiver vektoren med de tre variable på tid t, dvs. Yt =
(y1t; y2t; y3t) ; og hvor ?i = CiP og hvor P stammer fra en Cholesky decomponer- ing. Du bedes eksplicit rapportere responsen for y2 efter 3 perioder som følge af et chok til den første variabel.
?